app/jobs/subscription_placement_job.rb

```diff
@@ -28,29 +24,14 @@ def proxy_orders
   end
 
   def place_order_for(proxy_order)
-    JobLogger.logger.info("Placing Order for Proxy Order #{proxy_order.id}")
-    initialise_order(proxy_order)
-    place_order(proxy_order.order)
-  end
+    place_order = PlaceOrder.new(
+      proxy_order,
+      summarizer,
+      JobLogger.logger,
+      lambda { cap_quantity_and_store_changes(proxy_order.order) }
+    )
 
-  def initialise_order(proxy_order)
-    proxy_order.initialise_order!
-  rescue StandardError => e
-    Bugsnag.notify(e, subscription: proxy_order.subscription, proxy_order: proxy_order)
-  end
-
-  def place_order(order)
-    record_order(order)
-    return record_issue(:complete, order) if order.completed?
-
-    changes = cap_quantity_and_store_changes(order)
-    return handle_empty_order(order, changes) if order.line_items.where('quantity > 0').empty?
-
-    move_to_completion(order)
-    send_placement_email(order, changes)
-  rescue StandardError => e
-    record_and_log_error(:processing, order, e.message)
-    Bugsnag.notify(e, order: order)
+    place_order.call
   end
 
   def cap_quantity_and_store_changes(order)
```

Comment:
I purposefully left out the logic related to stock for another occasion so we focus only on that. It feels like this should be implemented elsewhere already :thinking:

Reply:
I'm afraid we'll have to stuff this stock-related logic into the new service for now. Although keeps the coupling and makes that service too bloated, it's not a responsibility of the job anymore.